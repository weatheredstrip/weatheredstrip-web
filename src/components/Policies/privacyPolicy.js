import React, { useState } from 'react';
import { Dialog, Paragraph, Link } from 'evergreen-ui'


const PP = () => (
  <>
    <Paragraph marginBottom={10}>Your privacy is important to us. It is Weathered Strip's policy to respect your privacy regarding any information we may collect from you across our website, <Link href="https://www.weatheredstrip.com">https://www.weatheredstrip.com</Link>, and other sites we own and operate.</Paragraph>
    <Paragraph marginBottom={10}>We only ask for personal information when we truly need it to provide a service to you. We collect it by fair and lawful means, with your knowledge and consent. We also let you know why we’re collecting it and how it will be used.</Paragraph>
    <Paragraph marginBottom={10}>We only retain collected information for as long as necessary to provide you with your requested service. What data we store, we’ll protect within commercially acceptable means to prevent loss and theft, as well as unauthorized access, disclosure, copying, use or modification.</Paragraph>
    <Paragraph marginBottom={10}>We don’t share any personally identifying information publicly or with third-parties, except when required to by law.</Paragraph>
    <Paragraph marginBottom={10}>Our website may link to external sites that are not operated by us. Please be aware that we have no control over the content and practices of these sites, and cannot accept responsibility or liability for their respective privacy policies.</Paragraph>
    <Paragraph marginBottom={10}>You are free to refuse our request for your personal information, with the understanding that we may be unable to provide you with some of your desired services.</Paragraph>
    <Paragraph marginBottom={10}>Your continued use of our website will be regarded as acceptance of our practices around privacy and personal information. If you have any questions about how we handle user data and personal information, feel free to contact us.</Paragraph>
    <Paragraph marginBottom={10}>This policy is effective as of 21 May 2020.</Paragraph>
    <Paragraph marginBottom={10}><Link href="https://getterms.io" title="Generate a free privacy policy">Privacy Policy created with GetTerms.</Link></Paragraph>
  </>
)

const PrivacyPolicyDialog = ({ isShown, onCloseComplete }) => (
  <Dialog
    title="Privacy Policy"
    isShown={isShown}
    hasCancel={false}
    confirmLabel="Close"
    onCloseComplete={onCloseComplete}
  >
    <PP />
  </Dialog>
)

const PrivacyPolicyDialogLink = () => {
  const [isShown, setShown] = useState(false)

  return (
    <>
      <PrivacyPolicyDialog isShown={isShown} onCloseComplete={() => setShown(false)}/>
      <Link href="#" onClick={() => setShown(true)}>Privacy Policy</Link>
    </>
  )
}


export { PrivacyPolicyDialog, PrivacyPolicyDialogLink }