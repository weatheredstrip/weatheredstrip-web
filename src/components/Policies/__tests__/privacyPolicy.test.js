import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import { PrivacyPolicyDialog, PrivacyPolicyDialogLink } from "../privacyPolicy";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders the full dialog text", () => {
  act(() => {
    render(<PrivacyPolicyDialog isShown={true} />, container);
  });

  const dialog = document.querySelector("[role=dialog]");

  expect(dialog.textContent).toMatchSnapshot();
});

it("renders the link for the dialog", () => {
  act(() => {
    render(<PrivacyPolicyDialogLink />, container);
  });
  expect(container).toMatchSnapshot();
})

it("opens the dialog when clicked", () => {
  act(() => {
    render(<PrivacyPolicyDialogLink />, container);
  });

  let dialog = document.querySelector("[role=dialog]");
  expect(dialog).toBeNull();

  // Get a hold of the Link
  const button = document.querySelector("[data-testid=privButton]");

  act(() => {
    button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });

  dialog = document.querySelector("[role=dialog]");
  expect(dialog).not.toBeNull();
})
