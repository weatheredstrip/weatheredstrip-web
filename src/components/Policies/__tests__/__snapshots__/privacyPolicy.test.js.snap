// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the full dialog text 1`] = `"Privacy PolicyYour privacy is important to us. It is Weathered Strip's policy to respect your privacy regarding any information we may collect from you across our website, https://www.weatheredstrip.com, and other sites we own and operate.We only ask for personal information when we truly need it to provide a service to you. We collect it by fair and lawful means, with your knowledge and consent. We also let you know why weâ€™re collecting it and how it will be used.We only retain collected information for as long as necessary to provide you with your requested service. What data we store, weâ€™ll protect within commercially acceptable means to prevent loss and theft, as well as unauthorized access, disclosure, copying, use or modification.We donâ€™t share any personally identifying information publicly or with third-parties, except when required to by law.Our website may link to external sites that are not operated by us. Please be aware that we have no control over the content and practices of these sites, and cannot accept responsibility or liability for their respective privacy policies.You are free to refuse our request for your personal information, with the understanding that we may be unable to provide you with some of your desired services.Your continued use of our website will be regarded as acceptance of our practices around privacy and personal information. If you have any questions about how we handle user data and personal information, feel free to contact us.This policy is effective as of 21 May 2020.Privacy Policy created with GetTerms.Close"`;

exports[`renders the link for the dialog 1`] = `
<div>
  <a
    class="css-1k92t60 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦txt-deco_underline ðŸ“¦box-szg_border-box"
    data-testid="privButton"
    href="#"
  >
    Privacy Policy
  </a>
</div>
`;
